File: src/creational/factory/python/classes/factories/car_factory.py
from classes.interfaces.vehicle_factory_protocol import VehicleFactoryProtocol
from classes.interfaces.vehicle_protocol import Vehicle
from classes.vehicles import Car


class CarFactory(VehicleFactoryProtocol):
    def get_vehicle(self, vehicle_name: str) -> Vehicle:
        return Car(vehicle_name)

File: src/creational/factory/typescript/classes/factories/CarFactory.ts
import { Vehicle } from '../interfaces/VehicleProtocol';
import { Car } from '../Vehicles';
import { VehicleFactoryProtocol } from './VehicleFactory';

export class CarFactory extends VehicleFactoryProtocol {
  getVehicle(vehicleName: string): Vehicle {
    return new Car(vehicleName);
  }
}
